Mathlib/CategoryTheory/Sums/Basic.lean

```diff
@@ -328,6 +351,155 @@ def symmetry : swap C D ⋙ swap D C ≅ 𝟭 (C ⊕ D) :=
 
 end Swap
 
+variable (A : Type u₁) [Category.{v₁} A] (A' : Type u₂) [Category.{v₂} A']
+  (B : Type u₃) [Category.{v₃} B]
+
+/-- The equivalence between functors from a sum and the product of the
+ functor categories. -/
+@[simps]
+def functorEquiv : A ⊕ A' ⥤ B ≌ (A ⥤ B) × (A' ⥤ B) where
+  functor :=
+    { obj F := ⟨inl_ A A' ⋙ F, inr_ A A' ⋙ F⟩
+      map η := ⟨whiskerLeft (inl_ A A') η, whiskerLeft (inr_ A A') η⟩ }
+  inverse :=
+    { obj F := Functor.sum' F.1 F.2
+      map η := NatTrans.sum' η.1 η.2 }
+  unitIso := NatIso.ofComponents <| fun F ↦ F.isoSum
+  counitIso := NatIso.ofComponents <| fun F ↦
+    (Functor.inlCompSum' _ _).prod (Functor.inrCompSum' _ _) ≪≫ prod.etaIso F
+
+variable {A A' B}
+
+@[simp]
+lemma functorEquiv_inverse_map_app_inl {X Y : (A ⥤ B) × (A' ⥤ B)} (η : X ⟶ Y) (a : A) :
+    ((functorEquiv A A' B).inverse.map η).app (inl a) = η.1.app a :=
+  rfl
```

Comment:
Some of these lemmas are already proven `by dsimp`, so that I do not think we need them. I would suggest removing those that can be proven `by dsimp` and keeping the others.